src/supervisor/watchers/handlers/index.ts

```diff
@@ -125,10 +136,105 @@ async function isSupportedWorkload(
   namespace: string,
   workloadKind: WorkloadKind,
 ): Promise<boolean> {
-  if (workloadKind !== WorkloadKind.DeploymentConfig) {
-    return true;
+  switch (workloadKind) {
+    case WorkloadKind.DeploymentConfig:
+      return await isDeploymentConfigSupported(namespace);
+    case WorkloadKind.CronJob:
+      return await isCronJobSupported(namespace);
+    case WorkloadKind.CronJobV1Beta1:
+      return await isCronJobV1Beta1Supported(namespace);
+    default:
+      return true;
+  }
+}
+
+async function isCronJobSupported(namespace: string): Promise<boolean> {
```

Comment:
I feel like there is a lot of repetition in these 3 functions, can we pass in the client and the workload kind into 1 function instead?